tests/unit/EmojiRegexTest.js

```diff
@@ -0,0 +1,42 @@
+import _ from 'underscore';
+import Emoji from '../../assets/emojis';
+import CONST from '../../src/CONST';
+import {isSingleEmoji} from '../../src/libs/ValidationUtils';
+
+describe('EmojiRegexTest', () => {
+    it('matches all the emojis in the list', () => {
+        const emojiMatched = _.every(Emoji, (emoji) => {
+            if (emoji.header === true || emoji.code === CONST.EMOJI_SPACER) {
+                return true;
+            }
+            const isEmojiMatched = isSingleEmoji(emoji.code);
+            let skinToneMatched = true;
+            if (emoji.types) {
+                skinToneMatched = _.every(emoji.types, emojiWithSkinTone => isSingleEmoji(emojiWithSkinTone));
+            }
+            return skinToneMatched && isEmojiMatched;
+        });
+
+        expect(emojiMatched).toBe(true);
+    });
+
+    it('matches single emojis variants for size', () => {
+        expect(isSingleEmoji('👨‍👩‍👦️')).toBe(true);
+        expect(isSingleEmoji('👉')).toBe(true);
+        expect(isSingleEmoji('👊🏾')).toBe(true);
+        expect(isSingleEmoji('😪️')).toBe(true);
+        expect(isSingleEmoji('❤️')).toBe(true);
+        expect(isSingleEmoji('👨‍👩‍👦️')).toBe(true);
+        expect(isSingleEmoji('😎️')).toBe(true);
+        expect(isSingleEmoji('🔫️')).toBe(true);
+        expect(isSingleEmoji('🛍')).toBe(true);
+        expect(isSingleEmoji('⁉️')).toBe(true);
+        expect(isSingleEmoji('*️⃣')).toBe(true);
+        expect(isSingleEmoji('✳️')).toBe(true);
+        expect(isSingleEmoji('👶🏽')).toBe(true);
+        expect(isSingleEmoji('🕍')).toBe(true);
+        expect(isSingleEmoji('🇺🇲')).toBe(true);
+        expect(isSingleEmoji('🇮🇳')).toBe(true);
+        expect(isSingleEmoji('🇺🇦️')).toBe(true);
```

Comment:
Is there any significance to checking all of these different emojis to see if they are single emoji? Could we just check one? Is there anything special about these ones?

Reply:
Well, I added one test for all the Emojis in our universe. These are some Emojis with different variations such as 2 length Emojis (flag), composite emoji (family), single emojis, text-based emoji (heart), etc.

Reply:
Ok haha I missed what the first test was doing. Could we maybe just mark next to some of these why we are specifically testing them (just the reasons you mentioned are fine). And thanks for making a test here 🙇

Reply:
and mean just add some comment like 

```js
        // GIVEN an emoji that has a length of 2 WHEN we check if it's a single emoji THEN it should return true
        expect(isSingleEmoji('🇺🇲')).toBe(true);
        expect(isSingleEmoji('🇮🇳')).toBe(true);
        expect(isSingleEmoji('🇺🇦️')).toBe(true);
```

Reply:
Maybe helpful to reference this also - https://github.com/Expensify/App/blob/main/tests/README.md#documenting-tests

We are not always following this rule but it would be nice to do so.

Reply:
Thanks for the share.  Will go through and update accordingly.

Reply:
Done